crates/modelardb_server/src/context.rs

```diff
@@ -296,38 +296,81 @@ impl Context {
             .deregister_table(table_name)
             .map_err(|error| ModelarDbError::TableError(error.to_string()))?;
 
-        // Drop the table from the storage engine by flushing the data managers. The table is
-        // marked as dropped in the data transfer component first to avoid transferring data to the
-        // remote data folder when flushing.
-        let storage_engine = self.storage_engine.write().await;
-        storage_engine.mark_table_as_dropped(table_name).await;
+        self.delete_table_from_storage_engine(table_name).await?;
 
-        storage_engine
-            .flush()
+        // Drop the table metadata from the metadata Delta Lake.
+        self.data_folders
+            .local_data_folder
+            .table_metadata_manager
+            .drop_table_metadata(table_name)
             .await
             .map_err(|error| ModelarDbError::TableError(error.to_string()))?;
 
-        storage_engine.clear_table(table_name).await;
+        // Drop the table from the Delta Lake.
+        self.data_folders
+            .local_data_folder
+            .delta_lake
+            .drop_delta_lake_table(table_name)
+            .await
+            .map_err(|error| ModelarDbError::TableError(error.to_string()))?;
+
+        Ok(())
+    }
+
+    /// Delete all data from the table with `table_name` if it exists. The table data is deleted
+    /// from the storage engine, metadata Delta Lake, and data Delta Lake. If the table does not
+    /// exist or if it could not be truncated, [`ModelarDbError`] is returned.
+    pub async fn truncate_table(&self, table_name: &str) -> Result<(), ModelarDbError> {
+        // Deleting the table from the storage engine does not require the table to exist, so the
+        // table is checked first.
+        if self.check_if_table_exists(table_name).await.is_ok() {
+            return Err(ModelarDbError::TableError(format!(
+                "Table with name '{table_name}' does not exist."
+            )));
+        }
+
+        self.delete_table_from_storage_engine(table_name).await?;
 
         // Delete the table metadata from the metadata Delta Lake.
         self.data_folders
             .local_data_folder
             .table_metadata_manager
-            .delete_table_metadata(table_name)
+            .truncate_table_metadata(table_name)
             .await
             .map_err(|error| ModelarDbError::TableError(error.to_string()))?;
 
-        // Delete the table from the Delta Lake.
+        // Delete the table data from the data Delta Lake.
         self.data_folders
             .local_data_folder
             .delta_lake
-            .drop_delta_lake_table(table_name)
+            .truncate_delta_lake_table(table_name)
             .await
             .map_err(|error| ModelarDbError::TableError(error.to_string()))?;
 
         Ok(())
     }
 
+    /// Delete the table from the storage engine by flushing the data managers and clearing the
```

Comment:
Maybe use drop instead of delete in the doc comment and method named for consistency. For example, a method that includes drop in its names is called by this method.

Reply:
Fixed.